client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -321,7 +322,12 @@ public void setupClient(OkHttpClient.Builder builder)
 
                 KnownTokenCache knownTokenCache = EXTERNAL_AUTHENTICATION_TOKEN_CACHE.getValue(properties).get();
 
-                ExternalAuthenticator authenticator = new ExternalAuthenticator(REDIRECT_HANDLER.get(), poller, knownTokenCache.create(), timeout);
+                RedirectHandler handler = EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS.getValue(properties)
+                        .map(CompositeRedirectHandler::new)
+                        .map(RedirectHandler.class::cast)
+                        .orElseGet(REDIRECT_HANDLER::get);
```

Comment:
Sorry, did we just break tests. Now the value is always present so `REDIRECT_HANDLER` won't be used at all.

Reply:
It seems so, but because default for EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS is ALL. We should probably make this other way around, so:

```
private static final AtomicReference<RedirectHandler> REDIRECT_HANDLER = new AtomicReference<>(null);


Optional.ofNullable(REDIRECT_HANDLER.get())
   .orElseGet(() -> EXTERNAL_AUTHENTICATION_REDIRECT_HANDLERS.getValue(properties)
      .map(CompositeRedirectHandler::new)
      .map(RedirectHandler.class::cast)
      .orElseThrow());
```

Reply:
can't do it this way. getValue() throws SQLException :)